pkg/client/listers/core/internalversion/node_expansion.go

```diff
@@ -1,48 +0,0 @@
-/*
-Copyright 2017 The Kubernetes Authors.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-package internalversion
-
-import (
-	"k8s.io/apimachinery/pkg/labels"
-	api "k8s.io/kubernetes/pkg/apis/core"
-)
-
-// NodeConditionPredicate is a function that indicates whether the given node's conditions meet
-// some set of criteria defined by the function.
-type NodeConditionPredicate func(node *api.Node) bool
-
-// NodeListerExpansion allows custom methods to be added to
-// NodeLister.
-type NodeListerExpansion interface {
-	ListWithPredicate(predicate NodeConditionPredicate) ([]*api.Node, error)
-}
-
-func (l *nodeLister) ListWithPredicate(predicate NodeConditionPredicate) ([]*api.Node, error) {
```

Comment:
Could this be used by external projects (other than Kubernetes itself)?

Reply:
@nikhita 😃I don't think so because the path containing "internal" cannot be imported by external projects.